fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -338,18 +345,54 @@ bool canLaunch(RelocateData& relocation,
 	return false;
 }
 
+bool canLaunchDest(const std::vector<std::pair<Reference<IDataDistributionTeam>, bool>>& candidateTeams,
+                   int priority,
+                   std::map<UID, Busyness>& busymapDest) {
+	// fail switch if this is causing issues
+	if (SERVER_KNOBS->RELOCATION_PARALLELISM_PER_DEST_SERVER <= 0) {
+		return true;
+	}
+	int workFactor = getDestWorkFactor();
+	for (auto& team : candidateTeams) {
+		for (UID id : team.first->getServerIDs()) {
+			if (!busymapDest[id].canLaunch(priority, workFactor)) {
+				return false;
```

Comment:
Why do we want to return `false` if any server on any team is overloaded? If there are some candidate teams that have capacity, shouldn't we still go ahead and launch?

Reply:
Ah yeah, so I remember being confused by this too at first when I was working on this.
Candidate teams is a vector because we may have multiple datacenters. The GetTeamRequest will return the best team per datacenter for the move to go to, it's not actually that we are considering multiple potential move destinations for a given shard in a given DC and choosing one.
Since this is confusing to both of us, it will likely be confusing to others. We may want to consider naming this better or adding comments to clarify.